tests/test_metadata.py

```diff
@@ -622,3 +624,137 @@ def test_disallowed_dynamic(self, field_name):
     def test_optional_defaults_to_none(self, field_name):
         meta = metadata.Metadata.from_raw({}, validate=False)
         assert getattr(meta, field_name) is None
+
+    @pytest.mark.parametrize(
+        ("license_expression", "expected"),
+        [
+            ("MIT", "MIT"),
+            ("mit", "MIT"),
+            ("BSD-3-Clause", "BSD-3-Clause"),
+            ("Bsd-3-clause", "BSD-3-Clause"),
+            (
+                "MIT AND (Apache-2.0 OR BSD-2-Clause)",
+                "MIT AND (Apache-2.0 OR BSD-2-Clause)",
+            ),
+            (
+                "mit and (apache-2.0 or bsd-2-clause)",
+                "MIT AND (Apache-2.0 OR BSD-2-Clause)",
+            ),
+            (
+                "MIT OR GPL-2.0-or-later OR (FSFUL AND BSD-2-Clause)",
+                "MIT OR GPL-2.0-or-later OR (FSFUL AND BSD-2-Clause)",
+            ),
+            (
+                "GPL-3.0-only WITH Classpath-exception-2.0 OR BSD-3-Clause",
+                "GPL-3.0-only WITH Classpath-exception-2.0 OR BSD-3-Clause",
+            ),
+            (
+                "LicenseRef-Special-License OR CC0-1.0 OR Unlicense",
+                "LicenseRef-Special-License OR CC0-1.0 OR Unlicense",
+            ),
+            ("mIt", "MIT"),
+            ("mit or apache-2.0", "MIT OR Apache-2.0"),
+            ("mit and apache-2.0", "MIT AND Apache-2.0"),
+            (
+                "gpl-2.0-or-later with bison-exception-2.2",
+                "GPL-2.0-or-later WITH Bison-exception-2.2",
+            ),
+            (
+                "mit or apache-2.0 and (bsd-3-clause or mpl-2.0)",
+                "MIT OR Apache-2.0 AND (BSD-3-Clause OR MPL-2.0)",
+            ),
+            ("mit and (apache-2.0+ or mpl-2.0+)", "MIT AND (Apache-2.0+ OR MPL-2.0+)"),
```

Comment:
nit: Missing some test cases for white-space normalization

Reply:
da3f04b322f23974e1bdc1d0fe820e3a35e85de8